hazelcast/src/main/java/com/hazelcast/internal/hotrestart/InternalHotRestartService.java

```diff
@@ -48,6 +50,21 @@ public interface InternalHotRestartService {
      */
     boolean isEnabled();
 
+    /**
+     * If recovery is in progress, stores the given {@code newClusterState} and
+     * applies it after recovery is complete, returning {@code true}. Otherwise
+     * does nothing and returns {@code false}.
+     * @param newClusterState
+     * @return {@code true} if recovery is in progress, indicating that the new cluster
+     * state will be applied once recovery is complete, otherwise {@code false}.
+     */
+    boolean setDeferredClusterState(ClusterState newClusterState);
+
+    /**
+     * @return {@code true} when recovery process is completed and all members reached final state, otherwise {@code false}.
```

Comment:
```suggestion
     * @return {@code true} when local recovery process is completed and all members reached final state, otherwise {@code false}.
```

Reply:
It's not about local recovery only; `startCompleted` becomes `true` after:
- local recovery from disk is done
- and `waitForDataLoadTimeoutOrFinalClusterStartStatus` completes (so either `CLUSTER_START_SUCCEEDED` -> all members succeeded or `CLUSTER_START_FAILED`).

Reply:
thx. maybe we can include this in javaDoc.